runtime/major_gc.c

```diff
@@ -394,9 +423,60 @@ void caml_add_to_orphaned_ephe_list(struct caml_ephe_info* ephe_info)
     ephe_info->must_sweep_ephe = 0;
     atomic_fetch_add_verify_ge0(&num_domains_to_ephe_sweep, -1);
   }
+  CAMLassert (ephe_info->live == 0);
+  CAMLassert (ephe_info->todo == 0);
```

Comment:
This comes from older code, but it would be more regular if we also had a `CAMLassert (ephe_info -> must_sweep_ephe == 0)` here, matching the three conditions on the function early exit.